test/Streamly/Test/Data/Fold.hs

```diff
@@ -444,6 +450,133 @@ headAndRest ls = monadicIO $ do
     taill [] = []
     taill (_:xs) = xs
 
+partitionByM :: Property
+partitionByM =
+    forAll (listOf1 (chooseInt (intMin, intMax)))
+        $ \ls0 -> monadicIO $ action ls0
+
+    where
+
+    action ls = do
+        let f = \x -> if odd x then return (Left x) else return (Right x)
+        v1 <- run $ S.fold (F.partitionByM f FL.length FL.length) $ S.fromList ls
+        let v2 = foldl (\b a -> if odd a then b+1 else b) 0 ls
+            v3 = foldl (\b a -> if even a then b+1 else b) 0 ls
+        assert (v1 == (v2, v3))
+
+partitionBy :: Property
+partitionBy =
+    forAll (listOf1 (chooseInt (intMin, intMax)))
+        $ \ls0 -> monadicIO $ action ls0
+
+    where
+
+    action ls = do
+        let f = \x -> if odd x then Left x else Right x
+        v1 <- run
+                $ S.fold (F.partitionBy f FL.length FL.length)
+                $ S.fromList ls
+        let v2 = foldl (\b a -> if odd a then b+1 else b) 0 ls
+            v3 = foldl (\b a -> if even a then b+1 else b) 0 ls
+        assert (v1 == (v2, v3))
+
+demux :: Expectation
+demux =
+    let table = Data.Map.fromList [("SUM", F.sum), ("PRODUCT", F.product)]
+        input = Stream.fromList
+                [ ("SUM", 1::Int)
+                , ("PRODUCT", 2::Int)
+                , ("SUM",3)
+                , ("PRODUCT", 4::Int)
+                ]
```

Comment:
Add a type signature to the entire list instead? `::Int` on one element is enough.

Reply:
Indentation